lib/cfg-grammar.y

```diff
@@ -924,15 +924,15 @@ options_item
 	;
 
 stat_option
-	: KW_STATS_FREQ '(' LL_NUMBER ')'          { last_stats_options->log_freq = $3; }
-	| KW_STATS_LEVEL '(' LL_NUMBER ')'         { last_stats_options->level = $3; }
-	| KW_STATS_LIFETIME '(' LL_NUMBER ')'      { last_stats_options->lifetime = $3; }
+	: KW_STATS_FREQ '(' nonnegative_integer ')'          { last_stats_options->log_freq = $3; }
+	| KW_STATS_LEVEL '(' nonnegative_integer ')'         { last_stats_options->level = $3; }
+	| KW_STATS_LIFETIME '(' positive_integer ')'      { last_stats_options->lifetime = $3; }
 	;
 
 dns_cache_option
-	: KW_DNS_CACHE_SIZE '(' LL_NUMBER ')'	{ last_dns_cache_options->cache_size = $3; }
-	| KW_DNS_CACHE_EXPIRE '(' LL_NUMBER ')'	{ last_dns_cache_options->expire = $3; }
-	| KW_DNS_CACHE_EXPIRE_FAILED '(' LL_NUMBER ')'
+	: KW_DNS_CACHE_SIZE '(' positive_integer ')'	{ last_dns_cache_options->cache_size = $3; }
+	| KW_DNS_CACHE_EXPIRE '(' nonnegative_integer ')'	{ last_dns_cache_options->expire = $3; }
```

Comment:
I think this is semantically incorrect. Is there any reason why you chose `nonnegative_integer`?

We have the following logic in the `dns_cache_lookup` function:
```
      if (entry->resolved &&
          ((entry->positive && entry->resolved < now - self->options->expire) ||
           (!entry->positive && entry->resolved < now - self->options->expire_failed)))
        {
          /* the entry is not persistent and is too old */
        }
```